web/packages/teleport/src/Navigation/SideNavigation/ResourcesSection.tsx

```diff
@@ -0,0 +1,349 @@
+      onClick: () => setPinnedUserPreference(true),
+    },
+    {
+      title: 'Applications',
+      icon: Icons.Application,
+      route: applicationsOnlyRoute,
+      searchableTags: ['resources', 'apps', 'applications'],
+      category: NavigationCategory.Resources,
+      exact: false,
+      customRouteMatchFn: () => isKindActive('app'),
+      onClick: () => setPinnedUserPreference(false),
+      subCategory: CustomNavigationSubcategory.FilteredViews,
+    },
+    {
+      title: 'Databases',
+      icon: Icons.Database,
+      route: databasesOnlyRoute,
+      searchableTags: ['resources', 'dbs', 'databases'],
+      category: NavigationCategory.Resources,
+      exact: false,
+      customRouteMatchFn: () => isKindActive('db'),
+      onClick: () => setPinnedUserPreference(false),
+      subCategory: CustomNavigationSubcategory.FilteredViews,
+    },
+    {
+      title: 'Desktops',
+      icon: Icons.Database,
+      route: desktopsOnlyRoute,
+      searchableTags: ['resources', 'desktops', 'rdp', 'windows'],
+      category: NavigationCategory.Resources,
+      exact: false,
+      customRouteMatchFn: () => isKindActive('windows_desktop'),
+      onClick: () => setPinnedUserPreference(false),
+      subCategory: CustomNavigationSubcategory.FilteredViews,
+    },
+    {
+      title: 'Kubernetes',
+      icon: Icons.Kubernetes,
+      route: kubesOnlyRoute,
+      searchableTags: ['resources', 'k8s', 'kubes', 'kubernetes'],
+      category: NavigationCategory.Resources,
+      exact: false,
+      customRouteMatchFn: () => isKindActive('kube_cluster'),
+      onClick: () => setPinnedUserPreference(false),
+      subCategory: CustomNavigationSubcategory.FilteredViews,
+    },
+    {
+      title: 'Servers',
+      icon: Icons.Server,
+      route: nodesOnlyRoute,
+      searchableTags: ['resources', 'servers', 'nodes', 'ssh'],
+      category: NavigationCategory.Resources,
+      exact: false,
+      customRouteMatchFn: () => isKindActive('node'),
+      onClick: () => setPinnedUserPreference(false),
+      subCategory: CustomNavigationSubcategory.FilteredViews,
+    },
+  ];
+}
+
+export function ResourcesSection({
+  expandedSection,
+  previousExpandedSection,
+  handleSetExpandedSection,
+  currentView,
+}: {
+  expandedSection: NavigationSection;
+  previousExpandedSection: NavigationSection;
+  currentView: NavigationSubsection;
+  handleSetExpandedSection: (section: NavigationSection) => void;
+}) {
+  const { clusterId } = useStickyClusterId();
+  const { preferences, updatePreferences } = useUser();
+  const section: NavigationSection = {
+    category: NavigationCategory.Resources,
+    subsections: [],
+  };
+  const baseRoute = cfg.getUnifiedResourcesRoute(clusterId);
+
+  const searchParams = new URLSearchParams(location.search);
+
+  const isExpanded = expandedSection?.category === NavigationCategory.Resources;
+
+  const subsections = getResourcesSubsections({
```

Comment:
> But I still feel that it's not ideal and we should treat resources section more like any other section.

I did explore this initially but the approach I went with is a bit cleaner imo. The `Resources` section needs to be treated differently than the "regular" sections because it doesn't have the same styling (for example it has a divider and "Filtered Views" text). Adding logic to handle these little differences based on category would unnecessarily complicate the `Navigation.tsx` component because this won't be a common pattern anyways, `Search` and `Resources` are the only custom "irregular" sections we plan on having, which is why I think it's fine to copy paste a bit of code for those cases and treat them modularly as opposed to using one single component with logic for conditional styling based on a particular category.